ceph-proxy/unit_tests/test_ceph_hooks.py

```diff
@@ -229,3 +229,43 @@ def test_update_apt_source(self, mock_package_install, mock_emit_cephconf):
         hooks.config_changed()
         mock_package_install.assert_called_with()
         mock_emit_cephconf.assert_called_with()
+
+    @mock.patch('ceph_hooks.is_leader')
+    @mock.patch('ceph_hooks.process_requests')
+    @mock.patch('ceph_hooks.ceph')
+    @mock.patch('ceph_hooks.ready')
+    def test_client_relation_changed(self, mock_ready, mock_ceph,
+                                     mock_process_requests, mock_is_leader):
```

Comment:
I don't think this test is overly complicated but ideally we'd have some commentary here as well what we want to test.

Reply:
Thanks for the review, added a bit more info